### PR TITLE
l10n_ch: QR-bill improvement

### PR DESCRIPTION
**Pull request by oco-odoo**
2 things in this PR:
- option to automatically set ISR number as payment reference on invoices (like done here in v13: https://github.com/odoo/odoo/commit/b49437371cec19197b9f4de391e26f9692d437c2)
- backport of the 13.3 stuff about Swiss QR-bills

### CODE REVIEW

**Comment by oco-odoo** [comment]
@fw-bot up to saas-13.2

**Comment by oco-odoo** [comment]
@yvaucher So you know it's coming ;)

File: addons/l10n_ch/models/res_bank.py [hunk 1]
```diff
@@ -86,75 +86,106 @@ def find_number(self, s):
...
-                    self.company_id.country_id.code and
-                    (t_street_comp != False) and
-                    (t_street_deb != False) and
-                    debitor.zip and
-                    debitor.city and
-                    debitor.country_id.code and
-                    (number != False) and (number_deb != False))
-        else:
-            return False
+    def build_swiss_code_url(self, amount, currency_name, not_used_anymore_1, debtor_partner, not_used_anymore_2, structured_communication, free_communication):
```
Comment (nim-odoo) [inline addons/l10n_ch/models/res_bank.py:89]: Changing the method signature will break the report if the user don't do a -u. You should use a different method name instead.

Comment (oco-odoo) [reply addons/l10n_ch/models/res_bank.py:89]: @nim-odoo Acutally, the parameters stay the same, I simply change their name to something more understandable (their content stays the same as before); so does it still count as a signature change? If so, why can it cause issues?

Comment (nim-odoo) [reply addons/l10n_ch/models/res_bank.py:89]: changing the name of parameters is fine :)

File: addons/l10n_ch/models/account_invoice.py [hunk 2]
```diff
@@ -185,3 +185,11 @@ def message_post(self, **kwargs):
         if self.env.context.get('l10n_ch_mark_isr_as_sent'):
             self.filtered(lambda inv: not inv.l10n_ch_isr_sent).write({'l10n_ch_isr_sent': True})
         return super(AccountInvoice, self.with_context(mail_post_autofollow=True)).message_post(**kwargs)
+
+    @api.multi
+    def _get_computed_reference(self):
+        self.ensure_one()
+        if self.company_id.invoice_reference_type == 'ch_isr':
```
[the remark this answers is not in the document]
Comment (oco-odoo) [reply addons/l10n_ch/models/account_invoice.py:192]: Yup, I kept a note about it; will do during the forward-port process :)

File: addons/l10n_ch/models/res_bank.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (oco-odoo) [reply addons/l10n_ch/models/res_bank.py:89]: I'm not breaking it, just renaming the parameters to make them more understandable (they keep the same content), and clearly mark the ones that aren't used anymore. All in all, it'll ease maintenance of this function ;)

File: addons/l10n_ch/models/res_bank.py [hunk 3]
```diff
@@ -86,75 +86,106 @@ def find_number(self, s):
...
+        return '/report/barcode/?type=%s&value=%s&width=%s&height=%s&humanreadable=1' % ('QR', werkzeug.urls.url_quote_plus('\n'.join(qr_code_vals)), 256, 256)
+
+    def _get_partner_address_lines(self, partner):
+        """ Returns a tuple of two elements containing the address lines to use
+        for this partner. Line 1 contains the street and number, line 2 contains
+        zip and city. Those two lines are limited to 70 characters
+        """
+        line_1 = (partner and partner.street or '') + ' ' + (partner and partner.street2 or '')
+        line_2 = partner.zip + ' ' + partner.city
+        return line_1[:71], line_2[:71]
```
[the remark this answers is not in the document]
Comment (oco-odoo) [reply addons/l10n_ch/models/res_bank.py:150]: Oops, right. Gonna change that in 13.3 as well.

**Comment by smetl** [comment]
@robodoo r+

**Comment by oco-odoo** [comment]
@robodoo rebase-ff

**Comment by smetl** [comment]
@robodoo rebase-ff andouille!

**Pull request closed without merging**